source/AI.h

```diff
@@ -118,31 +118,31 @@ template <class Type>
 	// "precision" is an optional argument corresponding to a value of the dot product of the current and target facing
 	// vectors above which no turning should be attempting, to reduce constant, minute corrections.
 	static double TurnToward(const Ship &ship, const Point &vector, const double precision = 0.9999);
-	static bool MoveToPlanet(Ship &ship, Command &command, double cruiseSpeed = 0.);
-	static bool MoveTo(Ship &ship, Command &command, const Point &targetPosition,
+	static bool MoveToPlanet(const Ship &ship, Command &command, double cruiseSpeed = 0.);
+	static bool MoveTo(const Ship &ship, Command &command, const Point &targetPosition,
 		const Point &targetVelocity, double radius, double slow, double cruiseSpeed = 0.);
-	static bool Stop(Ship &ship, Command &command, double maxSpeed = 0., const Point direction = Point());
-	static void PrepareForHyperspace(Ship &ship, Command &command);
-	static void CircleAround(Ship &ship, Command &command, const Body &target);
-	static void Swarm(Ship &ship, Command &command, const Body &target);
-	static void KeepStation(Ship &ship, Command &command, const Body &target);
-	static void Attack(Ship &ship, Command &command, const Ship &target);
-	static void AimToAttack(Ship &ship, Command &command, const Body &target);
-	static void MoveToAttack(Ship &ship, Command &command, const Body &target);
-	static void PickUp(Ship &ship, Command &command, const Body &target);
+	static bool Stop(const Ship &ship, Command &command, double maxSpeed = 0., Point direction = Point());
```

Comment:
I guess it would make sense to use a reference here, too.
```suggestion
	static bool Stop(const Ship &ship, Command &command, double maxSpeed = 0., const Point &direction = Point());
```

Reply:
I haven't changed any value parameters to const & yet.

I suggest doing that in a separate PR.
In particular passing Point by const & might be controversial as it looks like a fairly optimized value type.

Anyways, your call.

Reply:
We use `const Point &` pretty often